### PR TITLE
15

### LINKED ISSUE #15: Create Settings Page
https://www.figma.com/file/7SJDLLDVk7MVzN4zxx5mc6/First-Draft?node-id=28%3A3

Acceptance Criteria:
- Page looks like model in Figma

### PR DESCRIPTION
**Pull request by fredrikholmberg**
/closes #15

### CODE REVIEW

File: app/src/main/java/com/mobileappdevelopment/themunichquiz/MainActivity.kt [hunk 1]
```diff
@@ -7,6 +7,6 @@ class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContentView(R.layout.activity_settings)
```
Comment (lukasf98) [inline app/src/main/java/com/mobileappdevelopment/themunichquiz/MainActivity.kt:10]: This has to be reverted to activity_main
```suggestion
        setContentView(R.layout.activity_main)
```

File: app/src/main/java/com/mobileappdevelopment/themunichquiz/settings.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+
+class settings : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_settings)
+    }
+}
```
Comment (lukasf98) [inline app/src/main/java/com/mobileappdevelopment/themunichquiz/settings.kt:12]: Please see suggested changes for SettingsFragment.kt and remove this file.

File: app/src/main/java/com/mobileappdevelopment/themunichquiz/SettingsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+package com.mobileappdevelopment.themunichquiz
+
```
Comment (lukasf98) [inline app/src/main/java/com/mobileappdevelopment/themunichquiz/SettingsFragment.kt:2]:
```suggestion
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

class SettingsFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_settings, container, false)
        }
    }
```

File: app/src/main/res/layout/activity_settings.xml [hunk 4]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:aapt="http://schemas.android.com/aapt"
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".settings">
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:8]:
```suggestion
    tools:context=".SettingsFragment">
```

File: app/src/main/res/layout/activity_settings.xml [hunk 5]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:aapt="http://schemas.android.com/aapt"
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:2]:
```suggestion
<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

File: app/src/main/res/layout/activity_settings.xml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".settings">
+
+    <FrameLayout
+        android:id="@+id/frameLayout"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:13]:
```suggestion
```

File: app/src/main/res/layout/activity_settings.xml [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+        <ImageView
+            android:id="@+id/imageView12"
+            android:layout_width="499dp"
+            android:layout_height="754dp"
+            app:srcCompat="@drawable/backgroundimage" />
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:19]:
```suggestion
        <ImageView
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:src="@drawable/background"
        android:scaleType="centerCrop"
        android:contentDescription="@string/background" />
```

File: app/src/main/res/layout/activity_settings.xml [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+        <ImageView
+            android:id="@+id/imageView9"
+            android:layout_width="324dp"
+            android:layout_height="472dp"
+            app:srcCompat="@drawable/logo2" />
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:25]:
```suggestion
        <View
        android:id="@+id/logo"
        android:layout_width="323dp"
        android:layout_height="323dp"
        android:layout_marginTop="?actionBarSize"
        android:background="@drawable/logo"
        />
```

File: app/src/main/res/layout/activity_settings.xml [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+        <TextView
+            android:id="@+id/settings"
+            android:layout_width="410dp"
+            android:layout_height="65dp"
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:30]: Try to use relative constraints

File: app/src/main/res/layout/activity_settings.xml [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+            android:layout_gravity="center|top"
+            android:layout_marginLeft="154dp"
+            android:layout_marginTop="33dp"
+            android:gravity="top"
+            android:text="@string/settings"
+            android:textAppearance="@style/settings" />
+
+        <LinearLayout
+            android:layout_width="319dp"
+            android:layout_height="653dp"
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:42]: Also here, try use relative constraints like match_parent, wrap_content, etc.

File: app/src/main/res/layout/activity_settings.xml [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+                android:id="@+id/imageButton13"
+                android:layout_width="match_parent"
+                android:layout_height="39dp"
+                android:background="?android:attr/panelBackground"
+                app:srcCompat="@drawable/privacysettings" />
+
+            <ImageButton
+                android:id="@+id/imageButton14"
+                android:layout_width="match_parent"
+                android:layout_height="39dp"
```
Comment (lukasf98) [inline app/src/main/res/layout/activity_settings.xml:80]: Pleas use undestanable ids
```suggestion
            <Button
                android:id="@+id/button_help"
                android:layout_width="match_parent"
                android:layout_height="?actionBarSize"
```

File: app/src/main/res/values/styles.xml [hunk 12]
```diff
@@ -8,4 +8,16 @@
...
+        <item name="android:textSize">
+            18sp
+        </item>
+
+        <item name="android:textColor">
+            #000
+        </item>
+
+    </style>
+
```
Comment (lukasf98) [inline app/src/main/res/values/styles.xml:22]: Please check spacing and formats.
```suggestion
    <style name="settings">
            <item name="android:textSize">18sp</item>
            <item name="android:textColor">#000</item>
       </style>

```

File: app/src/main/AndroidManifest.xml [hunk 13]
```diff
@@ -9,6 +9,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
+        <activity android:name=".settings"></activity>
```
Comment (lukasf98) [inline app/src/main/AndroidManifest.xml:12]: I think you can remove this for now.
```suggestion
```

File: app/src/main/res/layout/activity_settings.xml (hunk 10 above)
Comment (fredrikholmberg) [reply app/src/main/res/layout/activity_settings.xml:42]: I tried but the picture doesn't fit then, that's why I've done it manually

Comment (lukasf98) [reply app/src/main/res/layout/activity_settings.xml:42]: Try this:
`<LinearLayout
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:layout_margin="20dp"
         android:layout_gravity="center"
         android:background="@drawable/rounded_corner"
         android:gravity="center"
         android:orientation="vertical">`

**Pull request closed without merging**